programs/fileio.c

```diff
@@ -698,9 +704,20 @@ static cRess_t FIO_createCResources(FIO_prefs_t* const prefs,
         CHECK( ZSTD_CCtx_setParameter(ress.cctx, ZSTD_c_rsyncable, prefs->rsyncable) );
 #endif
         /* dictionary */
-        CHECK( ZSTD_CCtx_setPledgedSrcSize(ress.cctx, srcSize) );  /* set the value temporarily for dictionary loading, to adapt compression parameters */
+        /* set the pledged size for dictionary loading, to adapt compression parameters */
```

Comment:
Actually this whole section doesn't do anything anymore.

Previously, the parameters you set at the time of dictionary loading effected the compression parameters used to construct the dictionary.

But, we don't want the order you set parameters and dictionaries to matter, so this has no effect anymore. So we only need the `loadDictionary()` call.

```c
CHECK( ZSTD_CCtx_loadDictionary(ress.cctx, dictBuffer, dictBuffSize) );
```

Reply:
We set the pledged src size for real here https://github.com/facebook/zstd/blob/096714d1b83f2e3950e88bc07f61c4f68f0e238b/programs/fileio.c#L1005.

Reply:
This comment still applies